src/message.rs

```diff
@@ -760,6 +763,10 @@ impl Message {
             .await
             .is_ok()
     }
+
+    pub fn get_error(&self) -> Option<String> {
```

Comment:
As this is the public Rust API (which we have to start building nicely at *some* point), this should come with an good doc comment describing what it is.  Like, when is there an error associated with a message, what kind of thing is this?